### PR TITLE
chore(ui-react): lint primitives

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Turn on linting for React primitives from `A*` to `M*`

#### Issue #, if available
NA

#### Description of how you validated changes
Ran unit tests and linted
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/primitives/types/autocomplete.ts [hunk 1]
```diff
@@ -144,3 +144,26 @@ export interface UseAutocompleteProps extends Partial<AutocompleteProps> {
 
   onClick?: React.MouseEventHandler<HTMLInputElement>;
 }
+
+type SetStateAction<T> = React.Dispatch<React.SetStateAction<T>>;
+
+export interface UseAutocomplete {
```
Comment (zchenwei) [inline packages/react/src/primitives/types/autocomplete.ts:150]: Maybe we should keep this stay in the same file as `useAutocomplete.tsx`?

File: packages/react/src/primitives/Icon/icons/IconAdd.tsx [hunk 2]
```diff
@@ -1,13 +1,14 @@
-import classNames from 'classnames';
 import * as React from 'react';
+import classNames from 'classnames';
```
Comment (zchenwei) [inline packages/react/src/primitives/Icon/icons/IconAdd.tsx:2]: We have a script to generate the icons. We need to update the script as well <- https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/scripts/generateIcons.js

File: packages/react/src/primitives/Icon/icons/types.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { ViewProps } from '../../types';
+
+export type InternalIcon = (
+  props: ViewProps & {
+    size?: string | number;
```
Comment (zchenwei) [inline packages/react/src/primitives/Icon/icons/types.ts:5]: I think we have remove `size` prop on Icon long time ago. Why is this required?

Comment (calebpollman) [reply packages/react/src/primitives/Icon/icons/types.ts:5]: There is usage of the `size` prop with the internal `Icon` components throughout our codebase

File: packages/react/src/primitives/types/autocomplete.ts (hunk 1 above)
Comment (calebpollman) [reply packages/react/src/primitives/types/autocomplete.ts:150]: I placed it with the rest of the `AutoComplete` component types for consistency. Don't feel strongly but is there a reason not include with the rest of the types?

Comment (zchenwei) [reply packages/react/src/primitives/types/autocomplete.ts:150]: No reason but looking at the `useCheckbox`, we group its type together

File: packages/react/src/primitives/Icon/icons/types.ts (hunk 3 above)
Comment (zchenwei) [reply packages/react/src/primitives/Icon/icons/types.ts:5]: I think we should clean up the use because it takes no effect.

Comment (calebpollman) [reply packages/react/src/primitives/Icon/icons/types.ts:5]: Are we certain? I see it used to handle `height` and `width` [here](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/src/primitives/Icon/icons/IconClose.tsx#L16-L17). That being said, think that it would be fine to make changes around `size` since it's all internal, but should probably be handled in a separate PR

File: packages/react/src/primitives/types/autocomplete.ts (hunk 1 above)
Comment (calebpollman) [reply packages/react/src/primitives/types/autocomplete.ts:150]: That's valid. I'm more apt to move `UseCheckbox` to the `Checkbox` types though

File: packages/react/src/primitives/Icon/icons/IconAdd.tsx (hunk 2 above)
Comment (calebpollman) [reply packages/react/src/primitives/Icon/icons/IconAdd.tsx:2]: Removed the script :) https://github.com/aws-amplify/amplify-ui/pull/3209

File: packages/react/src/primitives/Icon/icons/types.ts (hunk 3 above)
Comment (zchenwei) [reply packages/react/src/primitives/Icon/icons/types.ts:5]: Dang! I do not know why it was handled like this. That looks weird coz we should not lean on `size` any more and let the size scale based off parent font size

File: packages/react/package.json [hunk 4]
```diff
@@ -39,7 +39,7 @@
     "dev": "tsup --watch",
     "dev:build": "tsup",
     "clean": "rimraf dist node_modules",
-    "lint": "tsc --noEmit --project tsconfig.dev.json && yarn eslint ./src --ext .js,.ts,.tsx",
+    "lint": "tsc --noEmit --project tsconfig.dev.json && eslint",
```
Comment (calebpollman) [inline packages/react/package.json:42]: Forgot to not that this change speeds up the lint command by abt `166%` (from approximately 16 seconds to 6 seconds, YMMV)

**Review by zchenwei — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM
